docs/src/developers_guide/release.rst

```diff
@@ -257,6 +258,7 @@ Post Release Steps
 
 .. _SciTools/iris: https://github.com/SciTools/iris
 .. _tag on the SciTools/Iris: https://github.com/SciTools/iris/releases
+.. _GitHub Discussions forum: https://github.com/SciTools/iris/discussions
```

Comment:
Suggest we could link directly to the release category https://github.com/SciTools/iris/discussions/categories/releases using the text **GitHub Releases Discussion Forum**